src/app/lib/recipes/parser.ts

```diff
@@ -0,0 +1,65 @@
+import {BrewingRecipe, RecipeSchema} from "@/app/lib/definitions";
+import {roundAndTruncate} from "@/app/ui/utils";
+
+
+export const getBrewingRecipeUsingTotalWater = (rawRecipe: RecipeSchema, totalWaterAmount: number): BrewingRecipe => {
+  const coffeeAmount: number = (rawRecipe.coffeeGramsPerLiter * totalWaterAmount) / 1000
+  return getBrewingRecipe(rawRecipe, totalWaterAmount, coffeeAmount)
+}
+
+export const getBrewingRecipeUsingCoffeeAmount = (rawRecipe: RecipeSchema, coffeeAmount: number): BrewingRecipe => {
+  const totalWaterAmount: number = (coffeeAmount * 1000) / rawRecipe.coffeeGramsPerLiter
+  return getBrewingRecipe(rawRecipe, totalWaterAmount, coffeeAmount)
+}
+
+
+const evalRecipeSchemaTemplateString = (
+  templateString: string | undefined,
+  stepWaterAmountRaw: number,
+  waterAccumulatorRaw: number,
+  stepDuration: number
+): string => {
+  if (templateString == null) {
+    return ""
+  }
+  const flowRate: number = roundAndTruncate(stepWaterAmountRaw / stepDuration, 1)
+  const stepWaterAmount: number = roundAndTruncate(stepWaterAmountRaw, 1)
+  const waterAccumulator: number = roundAndTruncate(waterAccumulatorRaw, 1)
```

Comment:
I understand it's used by `eval`, rigth? Could you add a comment?

Reply:
yessss